### ISSUE TITLE
Standalone isomorphic url generator

### ISSUE THREAD

**Issue by fabiospampinato** [npmgraph/npmgraph-cli#1]
Hello, I'd like to make a little vscode extension for opening a project's `package.json` in npmgraph, and I'd like to reuse the logic that already exists for this in this module, but I don't really need a CLI, I just need a lower-level function that presumably receives the parsed `package.json` and returns a url.

Can something like this be implemented?

**Comment by broofa** [comment]
To be clear, what I hear you asking for is an SDK, a JS package that can be used to programmatically generate `npmgraph` URLs... right?

This is actually a good idea.  I'd be happy to take a look at a PR if you want to put something together.  I'd suggest you start by refactoring the code here to have two [workspaces](https://docs.npmjs.com/cli/v7/using-npm/workspaces): `packages/cli` and `packages/sdk` that can each be published as their own NPM package.

**Comment by fabiospampinato** [comment]
Yeah just a little npm package that exports a function for mapping packages into urls basically.

**Comment by fabiospampinato** [comment]
I just published the [extension](https://marketplace.visualstudio.com/items?itemName=fabiospampinato.vscode-open-in-npm-graph). In the end I could make do with something as simple as this:

```js
`https://npmgraph.js.org/?q=${encodeURIComponent ( `${pkg.name}@${pkg.version}` )}#packages=${encodeURIComponent ( JSON.stringify ( [pkg] ) )}`
```

**Comment by fregante** [comment]
I'd also use it to replace this code:

https://github.com/npmhub/npmhub/blob/908bfdbe1e275fc79f150404d3ddb976f68d19c0/source/components/HeaderLinkNpmGraph.svelte#L5-L11

I'm not sure if the extract complexity in this repo is worth it though. Maybe having a copy-pastable file would be good enough to simplify/expand this readme section: https://github.com/npmgraph/npmgraph#packages-hash-param-json-encoded

… or just publish a standalone package/repo, it's not like the format will change often.
